### ISSUE TITLE
bug: emphasis check style should not catch "strong" markup as well, or look into URLs

### ISSUE THREAD

**Issue by nnsite** [shinagawa-web/gomarklint#279]
With the new v3 latest version, I got a slew of consistent-emphasis-style errors. Cool. I have two criticisms which I think belong more to the bug category than features:
- currently, "emphasis" isn't only limited to *something* or _something_, but catches **something** and __something__ as well. This is unfortunate as I liked to differenciate visually emphasis from strong, and because my editor's highlighting handle _** but not ***. There should be two rules, one for each markup.
- it should not look into URLs like this one: [or sexual proactiveness](https://academia.edu/103656751/_Excess_and_Ergi_Exploring_the_Sexual_Valences_of_Supernatural_Episodes_in_the_%C3%8Dslendingas%C7%ABgur_ "Excess and Ergi: Exploring the Sexual Valences of Supernatural Episodes in the Íslendingasǫgur (2015)")

### Steps to reproduce

1. Create a Markdown file with the following content:
```md
[or sexual proactiveness](https://academia.edu/103656751/_Excess_and_Ergi_Exploring_the_Sexual_Valences_of_Supernatural_Episodes_in_the_%C3%8Dslendingas%C7%ABgur_ "Excess and Ergi: Exploring the Sexual Valences of Supernatural Episodes in the Íslendingasǫgur (2015)") 
**essai_strong** _inconsistent emphasis style_
```
2. Run: `gomarklint <file>`
3. See error / unexpected output

### Environment

- gomarklint version 3.0.5
- Linux 84-47-09-37-40-34 6.12.74+deb13+1-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.12.74-2 (2026-03-08) x86_64 GNU/Linux
- go version go1.24.4 linux/amd64

**Comment by shinagawa-web** [comment]
Fixed in v3.0.6. Both issues are resolved:

- `consistent-emphasis-style` no longer flags `**strong**`/`__strong__` and `*emphasis*`/`_emphasis_` against each other — they are now tracked independently.
- Underscores inside link destinations and titles (e.g. `[text](https://example.com/_path_ "title")`) are no longer mistaken for emphasis markers.
